static/app/views/insights/common/components/releaseSelector.tsx

```diff
@@ -134,8 +155,75 @@ function LabelDetails(props: LabelDetailsProps) {
   );
 }
 
+function getReleasesSortBy(
+  sort: ReleasesSortByOption,
+  environments: string[]
+): ReleasesSortByOption {
+  // Require 1 environment for date adopted
+  if (sort === ReleasesSortOption.ADOPTION && environments.length !== 1) {
+    return ReleasesSortOption.DATE;
+  }
+
+  if (sort in SORT_BY_OPTIONS) {
+    return sort;
+  }
+
+  // We could give a visual feedback to the user, saying that the sort by is invalid but
+  // since this UI will be refactored, maybe we just don't do anything now
+  return ReleasesSortOption.DATE;
```

Comment:
This makes sense but I think it's a little sketchy that both this _and_ `useReleases` specify a fallback, which might conflict!